runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -111,10 +112,20 @@ public void serialize(ByteBuf b) {
      * @return an empty list if object has no updates; a list of updates if exists
      */
     @Override
-    public List<SMREntry> getSMRUpdates(UUID id) {
-        MultiSMREntry entry = entryMap.get(id);
-        return entryMap.get(id) == null ? Collections.emptyList() :
-                entry.getUpdates();
+    public synchronized List<SMREntry> getSMRUpdates(UUID id) {
```

Comment:
Multiple VLOs can access this object at the same time, but only after it has been committed, so this should be thread-safe.